src/test/py/peer/peer_test.py

```diff
@@ -86,21 +87,28 @@ async def test_commission_art_piece(self, mock_input):
         This test verifies that the commission_art_piece method correctly adds a commission,
         publishes it on Kademlia, and schedules and sends a deadline notice.
         """
-        self.test_logger.debug(mock_input)
-        self.peer.node = self.mock_node
-        await self.peer.commission_art_piece()
-        self.assertEqual(len(self.peer.commissions), 1)
-        commission = self.peer.commissions[0]
-        self.assertEqual(commission.width, 10)
-        self.assertEqual(commission.height, 20)
-        self.assertLessEqual(commission.wait_time, timedelta(seconds=10))
-        self.mock_node.set.assert_called_with(
-            commission.get_key(), pickle.dumps(commission)
-        )
-        # Check that send_deadline_reached was called, which in turn calls our node's set method
-        self.mock_node.set.assert_called_with(
-            commission.get_key(), pickle.dumps(commission)
-        )
+        with patch(
+            "asyncio.get_event_loop",
+            return_value=MagicMock(
+                call_later=lambda *args: setattr(
+                    self, "deadline_task", asyncio.create_task(args[2])
+                )
+            ),
+        ):
+            self.test_logger.debug(mock_input)
+            self.peer.node = self.mock_node
+            await self.peer.commission_art_piece()
+            self.assertEqual(len(self.peer.commissions), 1)
```

Comment:
At some point later, a test should be added that tests the failure case and asserts that the commissions count does not increase.